evalml/automl/automl_search.py

```diff
@@ -403,6 +404,10 @@ def search(self, data_checks="auto", show_iteration_plot=True):
             show_iteration_plot (boolean, True): Shows an iteration vs. score plot in Jupyter notebook.
                 Disabled by default in non-Jupyter enviroments.
         """
+        if self._searched:
+            logger.info("AutoMLSearch has already been run and will not run again on the same instance. Re-initialize AutoMLSearch to search again.")
+            return
```

Comment:
@bchen1116 hmm after our discussion yesterday I thought we were going with option 3 [from the issue writeup](https://github.com/alteryx/evalml/issues/1424#issuecomment-746543694), not option 2. Did I remember wrong?

Reply:
@dsherry, I voiced for option 2 since the `X_train, y_train` data are now passed through the init of `AutoMLSearch`, and during the discussion yesterday, I believe everyone agreed, which is why I proceeded with option 2. Do you think option 3 would be better? My main issue with option 3 is that rerunning the same AutoMLClass, with the same data and defaults, should lead to the same results, and therefore isn't very useful. Let me know your thoughts, and we can also discuss more after standup if needed!

Reply:
That's entirely fair. Unless we provided a way for people to reconfigure the automl settings on a single automl search instance, rerunning the search from scratch should produce highly similar results. 👍 

Sounds good to me. Just wanted to make sure I wasn't missing something.